arbor/mechcat.cpp

```diff
@@ -608,7 +608,32 @@ const mechanism_catalogue& load_catalogue(const std::string& fn) {
      * different lifetime than the actual catalogue itfself. This is not a leak,
      * as `dlopen` caches handles for us.
      */
-    return *((const mechanism_catalogue*)get_catalogue());
+    int count = -1;
+    auto mechs = (arb_mechanism*)get_catalogue(&count);
+    if (count <= 0) {
+        throw bad_catalogue_error{util::pprintf("Invalid mechanism count {} in shared object '{}'", count, fn)};
+    }
+    mechanism_catalogue result;
+    for(int ix = 0; ix < count; ++ix) {
+        auto type = mechs[ix].type();
+        auto name = std::string{type.name};
+        if (name == "") {
+            throw bad_catalogue_error{util::pprintf("Empty name for mechanism in '{}'", fn)};
+        }
+        auto icpu = mechs[ix].i_cpu();
+        auto igpu = mechs[ix].i_gpu();
+        if (!icpu && !igpu) {
```

Comment:
This is enforcing a requirement that a mechanism must have _at least one_ implementation?

Reply:
Yes ... it's probably not strictly needed, but the sooner we reported these kind of issues...